schema/chat_room.proto

```diff
@@ -0,0 +1,343 @@
+option java_package = "com.google.showcase.v1alpha3";
+option java_multiple_files = true;
+
+option (google.api.metadata) = {
+  product_uri: "https://github.com/googleapis/gapic-showcase"
+};
+
+service ChatRoom {
+  rpc CreateRoom(CreateRoomRequest) returns (Room) {
+    option (google.api.http) = {
+      post: "/v1alpha3/rooms"
+      body: "*"
+    };
+    option (google.api.method_signature) = {
+      fields: ["display_name", "description"]
+    };
+  }
+
+  rpc GetRoom(GetRoomRequest) returns (Room) {
+    option (google.api.http) = {
+      get: "/v1alpha3/{name=rooms/*}"
+    };
+    option (google.api.method_signature) = {
+      fields: ["name"]
+    };
+  }
+
+  rpc UpdateRoom(UpdateRoomRequest) returns (Room) {
+    option (google.api.http) = {
+      patch: "/v1alpha3/{name=rooms/*}"
+      body: "*"
+    };
+    option (google.api.method_signature) = {
+      fields: ["name", "room.display_name"]
+      additional_signatures: {
+        fields: ["name", "room.display_name", "room.description"]
+      }
+    };
+  }
+
+  rpc DeleteRoom(DeleteRoomRequest) returns (google.protobuf.Empty) {
+    option (google.api.http) = {
+      delete: "/v1alpha3/{name=rooms/*}"
+    };
+    option (google.api.method_signature) = {
+      fields: ["name"]
+    };
+  }
+
+  rpc ListRooms(ListRoomsRequest) returns (ListRoomsResponse) {
+    option (google.api.http) = {
+      get: "/v1alpha3/rooms"
+    };
+  }
+
+  rpc StreamRoom(StreamRoomRequest) returns (stream Message) {
+    option (google.api.http) = {
+      post: "/v1alpha3/{name=rooms/*}:streamRoom"
+      body: "*"
+      additional_bindings: {
+        post: "/v1alpha3/{name=global}:streamRoom"
+        body: "*"
+      }
+    };
+  }
+
+  rpc SearchAll(SearchAllRequest) returns (google.longrunning.Operation) {
+    option (google.api.http) = {
+      post: "/v1alpha3:searchAll"
+      body: "*"
+    };
+    option (google.api.operation) = {
+      response_type: "google.showcase.v1alpha3.SearchAllResponse"
+      metadata_type: "google.showcase.v1alpha3.SearchAllMetadata"
+    };
+    option (google.api.method_signature) = {
+      fields: ["query"]
+    };
+  }
+
+  rpc CreateMessage(CreateMessageRequest) returns (Message) {
+    option (google.api.http) = {
+      post: "/v1alpha3/{parent=rooms/*}/messages"
+      body: "*"
+      additional_bindings: {
+        post: "/v1alpha3/{parent=global}/messages",
+        body: "*"
+      }
+    };
+    option (google.api.method_signature) = {
+      fields: ["parent", "message"]
+    };
+  }
+
+  rpc GetMessage(GetMessageRequest) returns (Message) {
+    option (google.api.http) = {
+      get: "/v1alpha3/{parent=rooms/*}/{name=messages/*}"
+      additional_bindings: {
+        get: "/v1alpha3/{parent=global}/name={messages/*}"
+      }
+    };
+  }
+
+  rpc UpdateMessage(UpdateMessageRequest) returns (Message) {
+    option (google.api.http) = {
+      patch: "/v1alpha3/{parent=rooms/*}/{name=messages/*}"
```

Comment:
This one has a slightly different correct answer:

```proto
option (google.api.http) = {
  patch: "/v1alpha3/{room.name=rooms/*/messages/*}"
};
```

Note the use of `room.name` instead of just `name`, which corresponds to `UpdateRoomRequest`'s only having `Room room`, and `name` is a sub-field of `room`.

Reply:
Makes sense. Done!